core/arch/arm/plat-sam/matrix.c

```diff
@@ -0,0 +1,610 @@
+/*
+ * Copyright (c) 2013, Atmel Corporation
+ * Copyright (c) 2017, Timesys Corporation
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the disclaimer below.
+ *
+ * Atmel's name may not be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * DISCLAIMER: THIS SOFTWARE IS PROVIDED BY ATMEL "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT ARE
+ * DISCLAIMED. IN NO EVENT SHALL ATMEL BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <arm32.h>
+#include <io.h>
+#include <sama5d2.h>
+#include <platform_config.h>
+#include <stdint.h>
+#include <matrix.h>
+#include <tz_matrix.h>
+#include <trace.h>
+
+#define MATRIX_AXIMX	1
+#define MATRIX_H64MX	2
+#define MATRIX_H32MX	3
+
+#define SECURITY_TYPE_AS	1
+#define SECURITY_TYPE_NS	2
+#define SECURITY_TYPE_PS	3
+
+struct peri_security {
+	unsigned int	peri_id;
+	unsigned int	matrix;
+	unsigned int	security_type;
+};
+
+static const struct peri_security peri_security_array[] = {
+	/*
+	 * AT91C_ID_1 - This is a undocumented bit in the datasheet.
+	 * However needs to be set for Linux to boot in "normal world"
+	 */
+	{
+		.peri_id = AT91C_ID_1,
+		.matrix = MATRIX_H64MX,
+		.security_type = SECURITY_TYPE_PS,
+	},
+	/* ARM */
```

Comment:
This comment (and all the others below) are kind of useless since they simply repeat what's written into `.peri_id`, they don't really improve readability either IMO. But you can keep them if you prefer.